client/core/trade.go

```diff
@@ -1542,6 +1655,7 @@ func (c *Core) tick(t *trackedTrade) (assetMap, error) {
 			redeems = append(redeems, match)
 			received += match.Quantity
 			quoteReceived += calc.BaseToQuote(match.Rate, match.Quantity)
+			t.updateDynamicSwapFeesPaid(c.ctx, match, false)
```

Comment:
Feels weird scheduling a fees watcher before we even successfully send the tx. Can this be reactive instead, more like `isSwappable` and `isRedeemable` which examine the current state to determine what actions are needed?

Reply:
If it blocks in the main tick loop causing more lock contention that's a no-go from me.

Reply:
Very good point though about it starting here before even sending the tx.  Seems more appropriate in swapMatchGroup and redeemMatchGroup.  Maybe I need to check out this branch and poke around.

Reply:
I don't think I've made this any more reactive. It is still queuing the check here. If I made `needsDynamicSwapFeeConfirms` to `dynamicSwapFeeConfirmed` and checked those off for all the coins it could be more like that. Let me know if that's the route I should go. The check later would then be pretty much the same plus determining if this is an initiation or redemption which we already know this way.